services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/HankeServiceImpl.kt

```diff
@@ -178,11 +180,35 @@ open class HankeServiceImpl(
     }
 
     @Transactional
-    override fun deleteHanke(hanke: Hanke, userId: String) {
-        hankeRepository.deleteById(hanke.id!!)
-        hankeLoggingService.logDelete(hanke, userId)
+    override fun deleteHanke(hankeId: Int, userId: String): Boolean {
```

Comment:
Why change this to take hankeId instead of hanke, when the the hanke is already fetched from DB in the controller?

Reply:
Some extra handling is required to make sure that the hanke exists, but this could all be avoided by taking in the actual non-nullable Hanke to begin with.

Reply:
Maybe change the Hanke to a HankeEntity, if that makes checking the applications easier?

Reply:
In the end this handles HankeEntity and the controller had Hanke (which does not know anything about hakemukset).

Reply:
I feel a bit uncomfortable using the entity in the controller. A better way would be to include hakemukset in Hanke object. But that may produce a lot more changes in the code base.

Reply:
Ok, I decided to change my approach quite a bit. Logic is now as follows:

1. Controller fetches a pair of Hanke and List<Application> from the service.
2. HankeService delete operation has hanke and applications as parameters now.

Why use a pair? Because Hanke does not contain information of its applications and controller should not be able to handle database entities. This current implementation also minimises database operations so it is better in that sense too.